docs/ExpectAPI.md

```diff
@@ -255,6 +283,91 @@ it('stores only 10 characters', () => {
 });
 ```
 
+#### Asymmetric matchers
+
+Asymmetric matchers are objects with an `asymmetricMatch` method with a single argument that returns `true` or `false`.  They can be passed as argument to some of the built in matchers, such as `toBeCalledWith` and `toEqual`, or to custom matchers if supported.
```

Comment:
It seems like this explains the implementation of what they are, but in the introduction here it would be better to explain what they are at a high level. What do you think about something like:

> Asymmetric matchers partially match values based on their character or structure instead of an exact value. This partial matching is called "asymmetric" because the assertion is not fully equal (symmetric) on both sides. Asymmetric matchers allow you to describe the parts of the value you care about, without needing to specify the entire value you're asserting. For example, if you only care that an object has a specific value but do not care about the other values in the object, so you can use the `objectContaining` asymmetric matcher as <example>.

Reply:
Yes that was what I was asking for in my previous comment

> Also given the usages of the asymmetricMatch function above there needs to be a clear definition of what an asymmetric matcher is.

'An asymmetric matcher participates in the matching process' ?
'An asymmetric matcher provides custom equality testing in the context of a matcher' ?

Reply:
What do you think of the suggestion I provided?

Reply:
@rickhanlonii Yes that's fine.  I am busy on my own work for a few days but I will definitely return to finish this pull request.